media-sound/supercollider: Add 3.13.0, Drop 3.11.2

andie-v opened the pull request:
Added ebuild for 3.13.0. 3.11.x no longer compiles due to an update in the libsndfile dependency. Removed four patches that are no longer needed in the current ebuild. Removed 3.11.2 ebuild.

Closes: https://bugs.gentoo.org/836355

andie-v commented:
I added myself as a maintainer but I'm still extremely new to all of this; I love supercollider and would enjoy keeping it working, but I can't necessarily say that I'm qualified for it. That being said, I'm eager to learn!!

Planning on bumping the ebuild file to EAPI 8, looking into the unused eclass, and looking into the NonexistantBlocker package to do a little more cleanup if possible.

andie-v commented:
correction for [7d2b88c]: app-admin/supernova seems to have been a package that conflicts with the 'supernova' binary this package adds since they have the same name. that package isn't in the repo so it seems like removing the blocker will be fine?

andie-v commented:
added another bugfix to get clang working, now i'm going to try and figure out why this won't compile with clang using the server USE flag...

andie-v commented:
rebased to drop unsigned the master-branch merge commit (whoops)

andie-v commented:
maybe capitalizing "Closes" will get it to link the bug?

sorry for the excessive comments

juippis commented on media-sound/supercollider/supercollider-3.13.0.ebuild:
```diff
@@ -0,0 +1,130 @@
+src_install() {
+	cmake_src_install
+
+	use emacs && newdoc editors/scel/README.md README.emacs
```
```
 * ERROR: media-sound/supercollider-3.13.0::gentoo failed (install phase):
 *   !!! newdoc: editors/scel/README.md does not exist
```

juippis reviewed (commented):
You should squash the follow-up commits to 3.13.0 to the one where you version bump it. So you'd have ~3 or so commits in total.
1: bump to 3.13.0 with all relevant changes
2: remove old files, and their .patch files
3: add yourself as maintainer.

andie-v commented:
vim has the same issue; support for both were split off into separate packages that are available through SC's package manager. fixed now.

switched over to xdg.eclass as suggested.

rebased as requested. guess i took "atomic commits" a little too far!!

andie-v commented:
dropping emacs made two more bugs irrelevant

juippis commented:
I wonder how comples they are? Do they require (heavy) compiling, or could they just be added into `SRC_URI` with a use flag, then dealt within this package? 

New packages that are dependencies of some old package are easier to get approved, since they're "needed".

andie-v commented:
ah wait I had a moment, sorry. i didn't know what submodules were (like I said, very new to all of this but very eager to learn), just saw that sc-el and scvim were in separate repos now. the real issue is that the scel directory should be called "sc-el". guessing the directory name got changed at some point? i spent an embarrassingly long amount of time trying to figure out why vim was still working and emacs wasn't.

thesamesam commented:
> @juippis - do you think I should add an einfo message telling users that emacs and vim were dropped? and those who wish to have extended functionality for those editors should seek out the relevant packages in SC's built-in package manager (Quarks) and/or build the plugins themselves?
> 
> i guess i could also make ebuilds for those two packages but it sounds like getting new packages accepted into the repo is a bit of a process

We're more happy (i.e. we treat it like it's a version bump) to have new packages if they're split out from an existing package. It's not the same as adding some new thing in its own right.

andie-v commented:
@juippis anything else I can do?

thesamesam commented:
I thought those extra packages needed adding in this PR?

andie-v commented:
oh hi no I'm just incompetent; they are still there in the form of submodules - see https://github.com/supercollider/supercollider/tree/develop/editors

I just wasn't aware what submodules were and made a bad assumption based on the fact that they appeared to be separate packages and thought that was the explanation behind the emacs USE flag breaking; in actuality the problem was just a result of the directory being changed from "scel" to "sc-el" upstream.

Sorry, I probably could've explained that better in my last comment!

thesamesam commented:
Thank you for explaining! Let me test this out now.

thesamesam commented on media-sound/supercollider/supercollider-3.13.0.ebuild:
```diff
@@ -12,7 +12,7 @@ SRC_URI="https://github.com/supercollider/supercollider/releases/download/Versio
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE="cpu_flags_x86_sse cpu_flags_x86_sse2 debug emacs +fftw gedit +gpl3 jack qt5 server +sndfile static-libs vim webengine X zeroconf"
+IUSE="cpu_flags_x86_sse cpu_flags_x86_sse2 debug emacs +fftw gedit +gpl3 jack qt5 server +sndfile static-libs vim webengine X +zeroconf"
```
We should probably install the vim files unconditionally per https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0301. It's unclear to me if Emacs is actually used at build time. If it isn't used to byte-compile anything, we should install those unconditionally too. Ditto gedit files.

But I'm not going to hold up this PR on that because it's totally broken right now in gentoo.git, and we want this working, so thank you!

andie-v replied:
Will do! Will start on that tomorrow for Vim as well as gedit (checked the CMakeLists and it doesn't compile anything and tested to confirm as well).

Looking at the [CMakeLists](https://github.com/supercollider/scel/blob/d8296a9af54d145703726328e22b9e89a08476af/el/CMakeLists.txt) file under scel/el, it looks like it compiles .el files and will throw an error if emacs isn't installed. Am I safe in assuming that probably means we shouldn't make it unconditional?

Speaking of emacs, I noticed the [readme](https://github.com/supercollider/scel/blob/d8296a9af54d145703726328e22b9e89a08476af/README.md) for it says this: 

> Whatever option you choose, make sure not to mix installation methods. In particular, do not install the Quark if you already have the supercollider-emacs package or if you compiled SuperCollider with the -DSC_EL=ON option. Otherwise you will get an error from SuperCollider about duplicated classes.

I'll try to put a check in for that in my next commit as well! That actually might fix [an open bug](https://bugs.gentoo.org/733496).

Thank you both for being so helpful!
